pkg/manager/manager.go

```diff
@@ -379,7 +387,17 @@ func New(config *rest.Config, options Options) (Manager, error) {
 		gracefulShutdownTimeout: *options.GracefulShutdownTimeout,
 		internalProceduresStop:  make(chan struct{}),
 		leaderElectionStopped:   make(chan struct{}),
-	}, nil
+	}
+
+	// A webhook server set by New's caller should be added now
+	// so GetWebhookServer can construct a new one if unset and add it only once.
+	if cm.webhookServer != nil {
+		if err := cm.Add(cm.webhookServer); err != nil {
```

Comment:
I'm not sure New() is the right place to be adding runnables? That seems like it will make testing complicated.

Reply:
I'm trying to think of a way to call `Add()` after `New()` without changing much. Perhaps some boolean flag to check/set in `GetWebhookServer()`.

Reply:
Yeah, I think some rearranging of the code in there would blend this more naturally with the existing flow.

Reply:
Decided to use `sync.Once` instead.